prompting-client/src/prompt_sequence.rs

```diff
@@ -95,14 +99,25 @@ impl TypedPromptFilter {
     }
 }
 
+/// Override the default handling for a missing key in being parsed as `None` so that only an
+/// explicit value of `null` is accepted.
+fn explicit_null<'de, T, D>(de: D) -> Result<Option<T>, D::Error>
```

Comment:
This is what allows us to distinguish the field being explicitly specified as `null` vs simply being omitted. (The default behaviour of `serde` when deserializing into an `Option<T>` field is to treat both the same).

This is adapted form the example shown [here](https://github.com/serde-rs/serde/issues/984#issuecomment-314143738) in order to emit an error in the case of a missing field rather than parsing into an `Option<Option<T>>` and having to handle the validation explicitly after the struct has been deserialized.